### PR TITLE
COOK-001C implemented CRUD and created recipe model [GP, PD, KS, ZD]

### LINKED ISSUE #22: COOK_001C Recipe overview: Controller

### PR DESCRIPTION
**Pull request by danielzirat**
How to start the application: ./mvnw spring-boot:run

### CODE REVIEW

**Comment by prdngr** [comment]
@danielzirat seems like there are some changes missing from your commit, please have a look at that.

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+@RestController
+public class RecipeController{
+
+  private final RepositoryHandler repository;
+
+  public RecipeController(RepositoryHandler repository) {
+    this.repository = repository;
+  }
+
+  @PostMapping("recipe/post")
```
Comment (tobiasrieger) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java]:
```suggestion
  @PostMapping("recipes")
```

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public Recipe addRecipe(@RequestBody Recipe newRecipe) {
+    return repository.save(newRecipe);
+  }
+
+  @GetMapping("recipe/get")
```
Comment (tobiasrieger) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java]:
```suggestion
  @GetMapping("recipes")
```

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  public List<Recipe> all() {
+    return repository.findAll();
+  }
+
+  // Single item
+  @GetMapping("recipe/get/{id}")
```
Comment (tobiasrieger) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java]:
```suggestion
  @GetMapping("recipes/{id}")
```

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  Recipe one(@PathVariable Long id) {
+
+    return repository.findById(id)
+            .orElseThrow(() -> new RecipeNotFoundException(id));
+  }
+
+  @PutMapping("recipe/update/{id}")
```
Comment (tobiasrieger) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java]:
```suggestion
  @PutMapping("recipes/{id}")
```

File: server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                recipe.setDescription(newRecipe.getDescription());
+                return repository.save(recipe);
+            })
+            .orElseGet(() -> {
+                newRecipe.setId(id);
+                return repository.save(newRecipe);
+            });
+  }
+
+  @DeleteMapping("recipe/delete/{id}")
```
Comment (tobiasrieger) [inline server/src/main/java/at/tugraz/asdmorning7/cook/controllers/RecipeController.java]:
```suggestion
  @DeleteMapping("recipes/{id}")
```

**Review by tobiasrieger — COMMENTED** [review]
Looks great, but please use restful routes!

**Review by prdngr — APPROVED** [review]

**Pull request merged**